packages/components/src/components/hds/popover-primitive/index.js

```diff
@@ -0,0 +1,245 @@
+import { guidFor } from '@ember/object/internals';
+import { modifier } from 'ember-modifier';
+
+import registerEvent from '../../../modifiers/hds-register-event';
+import anchoredPositionModifier from '../../../modifiers/hds-anchored-position';
+
+// https://github.com/oddbird/popover-polyfill?tab=readme-ov-file#with-npm
+// this is needed until Firefox releases the version 126 (up to 125 didn't support the Popover API)
+import {
+  // this call polyfills some of the browser methods to emulate the Popover API
+  apply as applyPopoverApiPolyfill,
+  // we'll use these two flags to overwrite the popover positioning strategy
+  // this is specifically done for Firefox (we need Firefox 126 to be released, to support the last 2 versions)
+  // see: https://whattrainisitnow.com/release/?version=126
+  isSupported as isPopoverApiSupported,
+  isPolyfilled as isPopoverApiPolyfilled,
+} from '@oddbird/popover-polyfill/fn';
+
+// we use this to re-export the values
+export {
+  PLACEMENTS,
+  DEFAULT_PLACEMENT,
+} from '../../../modifiers/hds-anchored-position';
+
+export default class HdsPopoverPrimitiveComponent extends Component {
+  @tracked isOpen = this.args.isOpen ?? false;
+  @tracked isClosing = false;
+  // this will enable "soft" events for the toggle ("hover" and "focus")
+  @tracked enableSoftEvents = this.args.enableSoftEvents ?? false;
+  // this will enable "click" events for the toggle
+  @tracked enableClickEvents = this.args.enableClickEvents ?? false;
+
+  constructor() {
+    super(...arguments);
+
+    // if the Popover API is not supported we need to polyfill it
+    if (!isPopoverApiSupported()) {
+      warn(
+        "The browser used does not support the Popover API so it's been emulated and some functionalities may not work as expected.",
+        {
+          id: 'hds-popover.no-popover-api-support.polyfill-applied',
+        }
+      );
+      // this function polyfills quite a few DOM methods and adds emulation for the Popover API
+      // see: https://github.com/oddbird/popover-polyfill/blob/main/src/popover.ts#L123
+      applyPopoverApiPolyfill();
+    }
+  }
+
+  setupPrimitiveContainer = modifier(
+    (element) => {
+      this.containerElement = element;
+
+      // we register the "soft" events
+      if (this.enableSoftEvents) {
+        registerEvent(this.containerElement, ['mouseenter', this.onMouseEnter]);
+        registerEvent(this.containerElement, ['mouseleave', this.onMouseLeave]);
+        registerEvent(this.containerElement, ['focusin', this.onFocusIn]);
+      }
+      // we always want the focusOut event
+      registerEvent(this.containerElement, ['focusout', this.onFocusOut]);
+    },
+    { eager: true }
+  );
+
+  setupPrimitiveToggle = modifier(
+    (element) => {
+      this.toggleElement = element;
+
+      assert(
+        `The toggle element of "Hds::PopoverPrimitive" must be a <button>; element received: <${element.tagName.toLowerCase()}>`,
+        element instanceof HTMLButtonElement
```

Comment:
Does it have to be a button tag? `<button type="button>Example HTML button</button>` or does it also work with `<input type="button" value="Example Input button" />`?

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/popoverTargetAction

Reply:
Technically yes, it supports also the `<input type="button" />` but I would avoid even suggesting the idea that they should use it.